src/components/ArticleCard.tsx

```diff
@@ -0,0 +1,38 @@
+import type { Article } from '@/types/article';
+import { Chip } from './Chip';
+import Image from 'next/image';
+import { formatToMonthDayKorean } from '@/utils/formatDate/formatToMonthDayKorean';
+
+interface ArticleCardProps extends Article {
+  isToday?: boolean;
+}
+
+const ArticleCard = (props: ArticleCardProps) => {
+  const { title, content, date, thumbnail, isRead, from, isToday = false } = props;
```

Comment:
props 이렇게 받은 이유가 있을까요!?

Reply:
원래 저도
```suggestion
const ArticleCard = ({title, content, date, ... , isToday = false } : ArticleCardProps) => {
...
```
와 같이 작업하는것을 선호했었는데, 어쩌다가 위의 코드와 같이 한줄로 쭉 받아오는게 좀 더 깔끔해보인다는 장점이 보였던 것 같아요
다만, 위의 props를 한줄에 몰아 받는 방식은 어떠한 props가 한줄에 쫙 펼쳐져있다보니 필요한 부분을 찾을 때 조금 불편한 단점이 사용하면서 있었습니다.

깔끔함보다는 props를 볼 때 필요한 부분을 한눈에 찾을 수 있는게 나으려나 하는 생각도 드는데 어떻게 생각하시나요?

Reply:
저도 `const ArticleCard = ({title, content, date, ... , isToday = false } : ArticleCardProps) => {` 와 같은 방법을 선호하긴 해서요 ㅎㅎ...
근데 선호와 별개로 코드 스타일적인 측면에서 전체적인 통일성에서 벗어나 새로운 스타일의 코드가 갑자기 나와서 여쭤봤습니다!
조금 길긴 한데 통일시키는게 좋지 않을까 생각이 드네요 ㅎㅎ

---

만약 프롭이 너무 많다 싶으면 아예 PropsWithChildren넣고 children 자체를 넣어버리는 방법도 있을 것 같아요! 
Card 자체를 백그라운드만 들어가도록 좀 더 공통컴포넌트화 시키구요.

그렇게 되면 `ArticleCard`라는 네이밍 대신 `Card`라는 네이밍으로 묶고, ArticleCardList자체로 퉁쳐서 컴포넌트화를 시킨 뒤, page컴포넌트에는 ArticleCardList만 넣는 방법도 있을 것 같아요!
너무 prop이 많아지니 굳이 프롭으로 안넘기고 확장성 없이 도메인에 종속시켜버리는거죠.

아니면 https://happysisyphe.tistory.com/70 이 글처럼 컴포넌트를 다 분리시키고, 응집성을 잘 고려하여 조합하는방식의 컴포넌트를 짜는 방법도 있을 것 같구요! 

물론 코드에 정답은 없지만 문기님은 어떻게 생각하실지 궁금하네요 ㅎㅎ

꼭 위 이야기들 반영하고 머지해달라는 이야기가 절대 아닙니다!

Reply:
일단 props는 모두 원래 하던대로 돌려놨습니다!
(진짜 정신없이 작업했나봐요 제가 전부 다 `const {...} = props` 방법으로 작업한줄알았는데
막상 한 5개 파일을 제외하고는 죄다 함수 선언부분에서 props를 받고있었네요 😢